### DISCUSSION TITLE
Turn on / off the JSONPath expression

### DISCUSSION THREAD

**Discussion by cspinetta** [Kong/insomnia#3851] [Ideas]
It would be nice and really useful to be able to temporarily disable a JSONPath expression, or just comment it (it could be using # or // as a lot of other programming language). When one write a JSONPath expression, it's common to need to remove it temporarily to inspect the whole JSON again (maybe to improve or modify the JSONPath expression).

**Comment by dimitropoulos** [comment]
any ideas on how this could be implemented? are you thinkin' like a button to click in the UI or something?  where would you see that button living?

**Reply by cspinetta** [reply]
Not sure what the level of effort that is involved, but I'm thinking like a checkbox on the left of the filter input box:

<img width="560" alt="insomnia" src="https://user-images.githubusercontent.com/3776757/127326427-1927c2dc-fa93-4a84-9540-9b39fcb0afdd.png">

I think it would be aligned with the style to disable query params, headers and so on.

**Comment by rodrigoscdc** [comment]
Other great feature should be to open the result of a JSONPath in other split panel, to see the full JSON response and JSONPath result. Other cool feature that is similar to this context would be to use https://jsonata.org/ in expressions too.
